spring-kafka/README.md

```diff
@@ -0,0 +1,357 @@
+# Spring Kafka
+
+## Goal
+The goal of this module is to be able to have a common code base to send and/or receive messages from a Kafka broker
+using Spring Kafka as the underlying library.
+
+This module also enforces that Google Protobuf is used as the message encoding.
+
+## Type Converter
+
+Through out the below examples we will use `type-converters` elements which are needed because the type erasure when injecting beans in 
+Karaf is not following the same algorithm as Java.
+Because of that methods which accept Generic types don't recognize the existing methods as valid ones to use in the injection.
+
+Further info can be seen in the `osgi-blueprint-api` module class [GenericsTypeConverter.java](https://github.com/pentaho/pentaho-osgi-bundles/blob/54c4e1d0ad1324962714d5c405318331b4f223f4/osgi-blueprint-api/src/main/java/org/hitachivantara/osgi/service/blueprint/GenericsTypeConverter.java)
+javadoc.
+
+See also below issues in apache Aries project to more details:
+[Apache ARIES-1500](https://issues.apache.org/jira/browse/ARIES-1500)
+[Apache ARIES-1607](https://issues.apache.org/jira/browse/ARIES-1607)
+
+## Producer
+
+To produce messages to Kafka topics the following blueprint definitions should be used:
+
+```xml
```

Comment:
Personally I prefer some overall "picture" of the things instead of jumping directly into the IoC declaration. This ties in with my previous comment of knowing and understanding the bigger picture.